### ISSUE TITLE
Question: Indentation control in {% include %}

### ISSUE THREAD

**Issue by gavrix** [kylef/Stencil#95]
Hi. I'm trying to use stencil for swift code generation (as many other existing applications of stencil) and found it very confusing when it comes to indentation control. It isn't very important for html code generation (as  jinja, liquid, django and many other implementations if this kind of templating usually do) because that html is then being interrupted by browsers and humans don't look in that geneted html. But for swift code generation it would be beneficial for generated code too look readable for humans too.

Simple example to describe the problem. Consider template
```
struct SomeType {
    {% include variablesSection%}
    {% include publicMethosSection %}
}
```

and then `publicMethosSection` too look like 
```
func method1() {
    doSomething()
}

func method2() {
    doSomething()
}
```

Master template assumes some indentation as both includes are prefixed with 4 spaces, however when rendering includes inner templates all new lines generated by inner template break that indentation.

In terms of templating language design, what is the suitable way of ensuring that all new lines coming from include gets automatically indented same way as `include` clause in master template? I understand this feature  may not exist yet, but how you think it may look like?

**Comment by yonaskolb** [comment]
I've been wanting this behaviour too. I guess the include block could prepend it's own indentation to every line of the included string. 
As a related but different issue, it would be useful to pass in an optional context to the included file, instead of it getting the whole master context

**Comment by yonaskolb** [comment]
This is fixed in #114
